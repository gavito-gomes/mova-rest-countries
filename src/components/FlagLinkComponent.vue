<template>
  <v-card :elevation="4">
    <router-link :to="href">
      <v-img :src="imgSrc" :aspect-ratio="16 / 9"></v-img>
    </router-link>
  </v-card>
</template>

<script>
export default {
  props: {
    imgSrc: {
      type: String,
      required: true,
    },
    href: {
      type: String,
      required: true,
    },
  },
}
</script>
