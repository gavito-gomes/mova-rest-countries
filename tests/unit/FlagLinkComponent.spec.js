import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import FlagLinkComponent from '@/components/FlagLinkComponent.vue'
import Vue from 'vue'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

describe('FlagLinkComponent.vue', () => {
  let vuetify

  const localVue = createLocalVue()
  localVue.use(VueRouter)
  const router = new VueRouter()

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('Renders a card with a link', () => {
    const wrapper = mount(FlagLinkComponent, {
      localVue,
      router,
      vuetify,
      propsData: { imgSrc: '', href: '' },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Receives href correctly', () => {
    const href = '#/details/BR'
    const wrapper = mount(FlagLinkComponent, {
      localVue,
      router,
      vuetify,
      propsData: { imgSrc: '', href },
    })

    const link = wrapper.find('a')

    expect(link.element.href).toMatch(href)
  })

  it('Redirect correctly when clicked', () => {
    const href = '/some/path'
    const wrapper = mount(FlagLinkComponent, {
      localVue,
      router,
      vuetify,
      propsData: { imgSrc: '', href },
    })

    const link = wrapper.find('a')

    link.trigger('click')

    expect(router.currentRoute.path).toBe(href)
  })
})
