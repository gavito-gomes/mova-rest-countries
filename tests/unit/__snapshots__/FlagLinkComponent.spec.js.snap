// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagLinkComponent.vue Renders a card with a link 1`] = `
<div class="v-card v-sheet theme--light elevation-4"><a href="#/" aria-current="page" class="router-link-exact-active router-link-active">
    <div class="v-image v-responsive theme--light">
      <div class="v-responsive__sizer" style="padding-bottom: 56.25%;"></div>
      <div class="v-responsive__content"></div>
    </div>
  </a></div>
`;
